### ISSUE TITLE
R-Z Polygon incorrectly rendered

### ISSUE THREAD

**Issue by hassec** [openmc-dev/openmc#2357]
Reproducer: 

```py
import numpy as np
import openmc as mc
import matplotlib.pyplot as plt
from matplotlib.patches import Polygon


rz_points = np.array(
    [
        [6.88, 3.02],
        [6.88, 2.32],
        [7.63, 0.0],
        [5.75, 0.0],
        [5.75, 1.22],
        [6.30, 1.22],
        [6.30, 3.02],
        [6.88, 3.02],
    ]
)

# plot matplotlib polygon
poly = Polygon(rz_points)
ax = plt.subplot()
ax.add_patch(poly)
ax.autoscale()
plt.savefig("mpl.png")


p0 = mc.model.Polygon(rz_points, basis="rz")
p0.boundary_type = "vacuum"

cell = mc.Cell(region=(-p0), name="my_cell")
geom = mc.Geometry([cell])

model = mc.Model()
model.geometry = geom
plot = mc.Plot()
plot.origin = (0, 0, 0)
plot.width = (20, 20)
plot.basis = "xz"
model.plots = [plot]
model.plot_geometry(False)
```

The matplotlib plot shows the shape of the polygon: 
![image](https://user-images.githubusercontent.com/11351671/213424594-20b29ae7-35fa-47e7-ad4d-5f93384d16cd.png)

But the geometry of the cell that is created by OpenMC doesn't correctly reproduce the top left cutout but has a diagonal instead.

![image](https://user-images.githubusercontent.com/11351671/213424787-1823db85-8664-44c6-bda0-2883c058f588.png)

fyi @eepeterson

**Comment by hassec** [comment]
Adding a plot of the delaunay triangulation: 
![image](https://user-images.githubusercontent.com/11351671/214304179-960c5c5c-12ae-487c-b5d5-7e186396e34c.png)

If I understand the current algorithm correctly it assumes that it can build the original polygon back from sets of convex subpolygons which are created by grouping the simplices created by the Delaunay triangulation. 

But, this doesn't seem to be possible for this specific case. 

Maybe a different approach to splitting the polygon into convex subsets, could work?

**Comment by eepeterson** [comment]
Ah, sorry for the delay - I've been away for a bit. The issue here is that the triangulation didn't produce an edge along one of the boundaries of the polygon. I'll see if there is an option in the QHull library to force a set of edges just from the input points. The polygon can always be decomposed into convex subsets as long as there are edges on the boundary of the polygon.

**Comment by eepeterson** [comment]
closed via #2362
